### PR TITLE
Added Support for SX1276 LoRa module for long range wireless communication

### PR DESCRIPTION
**Pull request by jithinbp**
+ added support for SX1276 LoRa module.  issue #62
+ tested the connectivity in an urban environment up to 500 meters.

### CODE REVIEW

**Comment by mariobehling** [comment]
@jithinbp Please don't merge your own commits. Please ask @CloudyPadmal to review. It is also a chance for everyone to learn more about the project.

It is important that devs follow best practices to give everyone a chance to align.

**Comment by jithinbp** [comment]
@mariobehling Apologies. I'll be including him for future PRs .
I already raised a new PR for a separate issue raised by @viveksb007 , so I don't know if I can undo this though.

File: PSL/SENSORS/Sx1276.py [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+	REG_PREAMBLE_LSB         = 0x21
+	REG_PAYLOAD_LENGTH       = 0x22
+	REG_MODEM_CONFIG_3       = 0x26
+	REG_RSSI_WIDEBAND        = 0x2c
+	REG_DETECTION_OPTIMIZE   = 0x31
+	REG_DETECTION_THRESHOLD  = 0x37
+	REG_SYNC_WORD            = 0x39
+	REG_DIO_MAPPING_1        = 0x40
+	REG_VERSION              = 0x42
+	REG_PA_DAC				 = 0x4D
```
Comment (CloudyPadmal) [inline PSL/SENSORS/Sx1276.py]: Alignments

Comment (jithinbp) [inline PSL/SENSORS/Sx1276.py]: Cool. i'm on it.
You can get the modules on AliExpress for as low as $4 - $5 .

Comment (CloudyPadmal) [inline PSL/SENSORS/Sx1276.py]: That's superb! :D I was checking the device in eBay. There are two types I found [`SPI`](http://www.ebay.com/itm/RCmall-2-4G-Wireless-Wifi-SPI-Transceiver-Module-with-Shield-Case-22dBm-100mW/182501827021?_trksid=p2047675.c100005.m1851&_trkparms=aid%3D555018%26algo%3DPL.SIM%26ao%3D2%26asc%3D41376%26meid%3D16c81dc3d550445e8d80c0214596cfd3%26pid%3D100005%26rk%3D2%26rkt%3D6%26sd%3D192002707560) and [`TTL`](http://www.ebay.com/itm/E44-TTL-100-20dBm-long-range-3km-SX1276-SX1278-915MHz-LoRa-RF-Transceiver-Module-/192002707560?hash=item2cb440d068:g:81IAAOSwZQRYbz8O)! Current implementation is using `SPI` only right?

Comment (jithinbp) [inline PSL/SENSORS/Sx1276.py]: As long as the underlying chip is SX1276, this library will work. I've used the 434 MHz modules from Dorji (DRF1276) as well as the RA-02 from AI-thinker. There are 868 MHz versions also, and only the right frequency needs to be selected in the code.

**Comment by jithinbp** [comment]
Thanks. Complete enough to manage P2P connectivity , but if you look at the massive documentation, it is actually a highly capable chip with highly advanced features.
Also, if you want to leverage existing commercial LoRaWAN gateways, one has to implement the entire LoRaWAN stack in Python. The LoRa modules from Microchip that use UART communication, and cost 5 times as much , have the LoRaWAN stack built into the firmware.

**Comment by hongquan** [comment]
You work is great. But next time could you please put more effort in coding style?

**Comment by jithinbp** [comment]
thanks. will do. I try to keep up with Codacy

**Pull request merged**